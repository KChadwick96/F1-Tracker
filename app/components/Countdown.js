import React from 'react';
import { View, Text, StyleSheet, Platform } from 'react-native';
import PropTypes from 'prop-types';

class Countdown extends React.Component {

  componentDidMount() {
    setInterval(() => {

    }, 1000);
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{this.props.title}</Text>
        <Text style={styles.countdown}>12D 5H 1M 10S</Text>
      </View>
    );
  }
}

Countdown.propTypes = {
  title: PropTypes.string.isRequired,
  countdownTo: PropTypes.object.isRequired
};

export default Countdown;

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
    paddingVertical: 20,
    marginTop: Platform.OS === 'ios' ? 25 : 0
  },
  title: {
    fontSize: 20,
    color: 'white'
  },
  countdown: {
    fontSize: 30,
    fontWeight: 'bold',
    color: 'white'
  }
});