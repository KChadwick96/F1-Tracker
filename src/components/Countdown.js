import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default ({title, countdownTo}) => {
  return (
    <View style={styles.container}>
      <Text>{title}</Text>
      <Text>12D 5H 1M 10S</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {

  }
});