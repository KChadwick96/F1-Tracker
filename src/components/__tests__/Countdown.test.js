// should take title and render
// should take date to countdown to

import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';

import Countdown from '../Countdown';

const component = <Countdown title="Next Race" />;

it('renders correctly', () => {
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
});

it('displays the title', () => {
    const clone = React.cloneElement(component);
    const wrapper = shallow(clone);
    expect(wrapper.find(Text).contains('Next Race')).toBeTruthy();
});

it('shows the countdown in the right format', () => {
    const clone = React.cloneElement(component);
    const wrapper = shallow(clone).render();

    console.log(wrapper);
    //console.log(wrapper.find(Text).last().);
    //expect(wrapper.find(Text).last().text).toBeTruthy();
});
