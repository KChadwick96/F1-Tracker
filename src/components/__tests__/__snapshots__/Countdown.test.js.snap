// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Next Race
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    12D 5H 1M 10S
  </Text>
</Component>
`;
